From: Thomas Bogendoerfer <tbogendoerfer@suse.de>
Date: Mon, 12 Jun 2017 14:54:57 +0200
Subject: net: phy: marvell: Show complete link partner advertising
Patch-mainline: v4.13-rc1
Git-commit: a1fa1a00b31c9155501d81f5396b2f6d76871fbe
References: bsc#1119113 FATE#326472

Give back all modes advertised by the link partner. This change brings
the marvell phy driver in line with all other phy drivers.

Signed-off-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
Reviewed-by: Florian Fainelli <f.fainelli@gmail.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---
 drivers/net/phy/marvell.c |    7 -------
 1 file changed, 7 deletions(-)

--- a/drivers/net/phy/marvell.c
+++ b/drivers/net/phy/marvell.c
@@ -1140,7 +1140,6 @@ static int marvell_read_status_page_an(s
 	int status;
 	int lpa;
 	int lpagb;
-	int adv;
 
 	status = phy_read(phydev, MII_M1011_PHY_STATUS);
 	if (status < 0)
@@ -1154,12 +1153,6 @@ static int marvell_read_status_page_an(s
 	if (lpagb < 0)
 		return lpagb;
 
-	adv = phy_read(phydev, MII_ADVERTISE);
-	if (adv < 0)
-		return adv;
-
-	lpa &= adv;
-
 	if (status & MII_M1011_PHY_STATUS_FULLDUPLEX)
 		phydev->duplex = DUPLEX_FULL;
 	else
